svf/lib/SVFIR/SVFIR.cpp

```diff
@@ -468,7 +469,7 @@ NodeID SVFIR::addFIObjNode(const MemObj* obj)
     //assert(findPAGNode(i) == false && "this node should not be created before");
     NodeID base = obj->getId();
     memToFieldsMap[base].set(obj->getId());
-    BaseObjVar*node = new BaseObjVar(obj->getValue(), obj->getId(), obj);
+    BaseObjVar* node = new BaseObjVar(obj->getValue(), obj->getId(), obj->getObjTypeInfo(), obj->getGNode());
```

Comment:
why we need to pass the parent class object "obj->getGNode()" here to create a subclass object?